src/main/java/io/netty/incubator/codec/quic/HmacSignQuicConnectionIdGenerator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.incubator.codec.quic;
+
+import java.nio.ByteBuffer;
+import java.security.SecureRandom;
+import java.util.Arrays;
+
+import io.netty.util.internal.ObjectUtil;
+import org.apache.commons.codec.digest.HmacAlgorithms;
+import org.apache.commons.codec.digest.HmacUtils;
+
+/**
+ * A {@link QuicConnectionIdGenerator} which creates new connection id by signing the given input
+ * using hmac algorithms
+ */
+final class HmacSignQuicConnectionIdGenerator implements QuicConnectionIdGenerator {
+    static final QuicConnectionIdGenerator INSTANCE = new HmacSignQuicConnectionIdGenerator();
+    private static final byte[] randomKey = new byte[16];
+
+    static {
+        new SecureRandom().nextBytes(randomKey);
+    }
+
+    private HmacSignQuicConnectionIdGenerator() {
+    }
+
+    @Override
+    public ByteBuffer newId(int length) {
+        throw new UnsupportedOperationException();
```

Comment:
Why is this not supported ?
```suggestion
        return newId(Buffer.allocate(0), length);
```

Reply:
I think a signIdGenerator should have an input to sign with, `newId(Buffer.allocate(0), length)` will generate the same id which I think can cause some unpredictable issue when we use it in our project. server should always use `ByteBuffer newId(ByteBuffer input, int length)`